### PR TITLE
Add script to plot detected objects

### PR DESCRIPTION
**Pull request by adamlm**
# PR Details
## Description

This PR adds a script to plot detected objects' positions as detected from Platform and as reported by incoming SDSMs.

## Related Jira Key

[CDAR-846](https://usdot-carma.atlassian.net/browse/CDAR-846)

## Motivation and Context

Verification test metric

## How Has This Been Tested?

Manually with simulation data

## Types of changes

- [x] New feature (non-breaking change that adds functionality)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.

[CDAR-846]: https://usdot-carma.atlassian.net/browse/CDAR-846?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

People Tracking:
@MishkaMN

### CODE REVIEW

File: vru_use_case_scripts/plot_detected_objects [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    )
+
+    position_y.plot(
+        object_sdsm["cdasim_time_ms"] / 1e3,
+        object_sdsm["y"] + sensor_position["y"],
+        label="SDSM"
+    )
+
+    position_y.set_ylabel("$y$ (map) [m]")
+    # position_y.legend()
```
Comment (paulbourelly999) [inline vru_use_case_scripts/plot_detected_objects:239]: Is this still required

File: vru_use_case_scripts/plot_detected_objects [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+    position_x.set_ylabel("$x$ (map) [m]")
+    # position_x.legend()
+    lgd = position_x.legend(bbox_to_anchor=(1.04, 1), borderaxespad=0)
+
+    det_tck_x = splrep(object_detection_unique_times["cdasim_time_ms"].tolist(), object_detection_unique_times["map_position_x_m"].tolist(), k=5)
+    sdsm_tck_x = splrep(object_sdsm_unique_times["cdasim_time_ms"].tolist(), (object_sdsm_unique_times["x"] + sensor_position["x"]), k=5)
+
+    error_x.plot(times / 1e3, abs(BSpline(*det_tck_x)(times) - BSpline(*sdsm_tck_x)(times)))
+    error_x.set_ylabel("Error [m]")
```
Comment (paulbourelly999) [inline vru_use_case_scripts/plot_detected_objects:224]: One of these plots is for y error and the other for x error. Should we include this information in the y axis label or in a plot title or something.

File: vru_use_case_scripts/plot_detected_objects [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+from matplotlib import pyplot as plt
+import numpy as np
+import pandas as pd
+from scipy.interpolate import splrep, BSpline
+
+
+def parse_positions_from_sdsm_csv(file_path: Path):
+    def get_timestamp(csv_row: list):
+        return round(float(csv_row[1]))
```
Comment (paulbourelly999) [inline vru_use_case_scripts/plot_detected_objects:18]: Is this a problem with my  parsing script that should get fixed.

File: vru_use_case_scripts/plot_detected_objects [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+    def get_json_payload(csv_row: list):
+        json_string = csv_row[8]
```
Comment (paulbourelly999) [inline vru_use_case_scripts/plot_detected_objects:21]: I assume we are taking indexes to avoid column name changes impacting this logic? Is that accurate?

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Mentioned some small things. Also the plot looks a little like the time sync is slightly off although that code be delay. I thought the delay was on the scale of like .1-.3 seconds and this looks a little larger but that could be subjective.

**Comment by adamlm** [comment]
> Mentioned some small things. Also the plot looks a little like the time sync is slightly off although that code be delay. I thought the delay was on the scale of like .1-.3 seconds and this looks a little larger but that could be subjective.

The time delay compensation depends on the `parse_ros_bags.py` script. If there are issues with the value, they should be addressed in a separate issue.

File: vru_use_case_scripts/plot_detected_objects (hunk 1 above)
Comment (adamlm) [reply vru_use_case_scripts/plot_detected_objects]: No. I removed it.

File: vru_use_case_scripts/plot_detected_objects (hunk 3 above)
Comment (adamlm) [reply vru_use_case_scripts/plot_detected_objects:18]: I did this for correctness. The simulation time values from the `sdsm.csv` file are floating-point, but they look like `123.000045353`, `124.000949531`, _etc._

Comment (paulbourelly999) [reply vru_use_case_scripts/plot_detected_objects:18]: We should probably fix that in a separate issue. Probably won't happen now but at some point in the future.

File: vru_use_case_scripts/docs/plot_detected_objects_example.png
Comment (paulbourelly999) [inline vru_use_case_scripts/docs/plot_detected_objects_example.png:1]: Nit: This example looks like its before you made the changes to the y labels for the error plots.

**Review by paulbourelly999 — APPROVED** [review]
Looks good to me. Left one nit picky comment. Address at your discretion.

**Pull request merged**